crates/sui-framework/packages/sui-framework/sources/crypto/nitro_attestation.move

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::nitro_attestation;
+
+use sui::clock::{Self, Clock};
+
+public struct NitroAttestationDocument has store, copy, drop  {
+    module_id: vector<u8>,
+    timestamp: u64,
+    digest: vector<u8>,
+    pcrs: vector<vector<u8>>,
+    public_key: Option<vector<u8>>,
+    user_data: Option<vector<u8>>,
+    nonce: Option<vector<u8>>,
+}
+
+/// Internal native function
+native fun verify_nitro_attestation_internal(
+    attestation: &vector<u8>,
+    current_timestamp: u64
+): NitroAttestationDocument;
+
+/// @param attestation: attesttaion documents bytes data. 
+/// @param clock: the clock object.
+///
+/// Returns parsed NitroAttestationDocument after verifying the attestation.
+public fun verify_nitro_attestation(
```

Comment:
```suggestion
entry fun load_nitro_attestation(
```

Reply:
iiuc this forbids other move module to call this? i would think its useful to expose it?

Reply:
yes, other modules cannot call it, but in which use case do we need that? it's a user provided data.

as an entry function we have the flexibility to change the interface later if we like

Reply:
its user provided data, but the output are consumed by the caller module which should check the rest like verifying the public key == user data and the signature verifies, and all pcrs match the registry? 

https://github.com/MystenLabs/enclave-twitter/blob/main/move/sources/twitter.move#L25

Reply:
that function could receive NitroAttestationDocument as an output a previous PTB command